verbit/streaming_client.py

```diff
@@ -554,9 +614,15 @@ def _reconnect_generator(self, response_generator) -> typing.Iterator[typing.Dic
             except self.CONNECTION_EXCEPTION_CLASSES as connection_error:
                 self._log_exception(f'Error while generating responses', connection_error)
 
-                # wait for other threads, that still access the web-socket to fail and stop
+                # wait for other threads, that still access the WebSocket to fail and stop
                 self._wait_for_thread(timeout_step=0.1, global_timeout=1.0)
 
+                # if media stream already finished
+                if self._media_stream_finished:
+                    self._logger.warning('Media stream already finished! '
+                                         'Will not attempt to reconnect to WebSocket as server will not return any responses.')
+                    return
+
                 # try reconnecting and keep on yielding from the same generator
                 self._logger.debug('Trying to reconnect')
                 response_generator = super().start_stream(self._media_generator, self._media_config, self._response_types)
```

Comment:
In case it's still deprecated, let's not call the deprecated method `super().start_stream()` from here.
Especially that by signature it's incorrect in case of a `None` generator... so either `super()._start_stream()`, or no change in case the deprecation is removed.

Reply:
That was actually a bug! Good catch! 💪🏽 
I changed it to `_connect_and_start`.